Automattic/jetpack
# Subscriptions: fix comment subscription toggle behavior

Fixes #18045 and #10047

#### Changes proposed in this Pull Request:
There are two comment subscription settings. One setting enables subscriptions to new posts, and one enables subscriptions to new comments. These settings are maintained using the options`stb_enabled` and `stc_enabled`.

When these options do not exist, the settings default to 'on'. However, because of the way that non-existent options are handled, the statuses of the toggle switches for these settings do not currently match the behavior, as described in #18045.

Two changes are necessary to fix this:

 * When a comment subscription setting is disabled, convert the false setting value to 0. This change will allow the option value to be updated when the option doesn't exist yet.

 * Remove the module name argument from the calls to `getOptionValue()` in `SubscriptionsComponent`. Currently, if the subscriptions module hasn't been enabled yet, the values of the `stb_enabled` and `stc_enabled` options are undefined, and the toggles display as 'off'.
 
    The default status of these settings is 'on', and the toggles should display the actual status of the settings. Removing the module name argument from the `getOptionValue()` calls allows the status of the settings to be displayed correctly, regardless of whether the subscriptions module is enabled.

#### Jetpack product discussion
* n/a

#### Does this pull request change what data or activity we track or use?
* no

#### Testing instructions:

1. Create a new site. Install this branch. Activate, and connect Jetpack.
2. Navigate to `Jetpack -> Settings -> Discussion`. Do not enable Jetpack comments. These changes affect the WP comments form.
3. Enable the Subscriptions `Let visitors subscribe to new posts and comments via email` setting. Notice that the two sub-settings are enabled.
4. In a separate tab, navigate to a post. Notice that the comments form has two checkboxes, as expected:
    * Notify me of follow-up comments by email.
    * Notify me of new posts by email.
5. Go back to the Jetpack settings in wp-admin. Disable and enable the subscriptions settings, and verify that the checkboxes on the post display as expected. Also, reload the settings page and verify that the settings do not change.
6. In Calypso, navigate to `Settings -> Discussion`. Scroll down to the Subscriptions settings. Enable and disable these settings, and verify that the checkboxes on the post display as expected. Verify that no error messages are generated. Also, reload the settings page and verify that the settings do not change.

#### Proposed changelog entry for your changes:
* Subscriptions: fix the behavior of the toggle switches for the comment subscription settings.
